admin/services/pupil.service.js

```diff
@@ -73,4 +58,46 @@ pupilService.validatePupil = async (pupil, pupilData) => {
   return true
 }
 
+/**
+ * Fetch pupil's status
+ * @param pupil
+ * @returns {String}
+ */
+pupilService.getStatus = async (pupil) => {
```

Comment:
this (along with the school controller) need breaking up.  This could have gone in a separate service

Reply:
I thought about it as well. I'll generate a pupil.status.service and extract it completely.